File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        lo = 1
        hi = max(piles)
        while lo <= hi:
            k = (hi + lo) // 2
            kSum = sum(math.ceil(pile / k) for pile in piles)
            if kSum <= h:
                hi = k - 1
            elif kSum > h:
                lo = k + 1
        return lo

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int lo = 1;
        int hi = Arrays.stream(piles).max().getAsInt();
        while (lo <= hi) {
            int k = (hi - lo) / 2 + lo;
            int kSum = 0;
            for (int pile : piles) {
                kSum += Math.ceil((double) pile / k);
            }
            if (kSum <= h) {
                hi = k - 1;
            } else if (kSum > h) {
                lo = k + 1;
            }
        }
        return lo;
    }
}
